src/rdkafka_broker.c

```diff
@@ -5238,20 +5267,36 @@ int rd_kafka_brokers_add0(rd_kafka_t *rk, const char *brokerlist) {
                         break;
 
                 rd_kafka_wrlock(rk);
+                if (rk->rk_conf.enable_bootstrap_servers_canonical_resolve) {
```

Comment:
Shouldn't this logic should be in resolve function? We are just adding host names here.